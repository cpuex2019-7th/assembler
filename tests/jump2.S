label1:
    addi t1, zero, 0            ; 0
    jal a0, label3              ; 4
label2:
    add t1, t1, a0              ; 8
    jal a0, ret                 ; 12
label3:
    add t1, t1, a0              ; 16
    jal a0, label2              ; 20
ret:
    add a0, t1, a0              ; 24
