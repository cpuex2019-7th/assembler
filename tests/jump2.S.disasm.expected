addi t1, zero, 0
jal a0, 12 ; jumps to 00000010
add t1, t1, a0
jal a0, 12 ; jumps to 00000018
add t1, t1, a0
jal a0, -12 ; jumps to 00000008
add a0, t1, a0
